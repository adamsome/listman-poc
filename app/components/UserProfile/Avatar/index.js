import React from 'react'

const UserProfileAvatar = ({ username, imgSrc }) => (
  <p className="image is-128x128">
    <img
      alt={`${username} Avatar`}
      src={imgSrc}
    />
  </p>
)

export default UserProfileAvatar

