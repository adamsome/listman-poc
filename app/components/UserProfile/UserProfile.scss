.title.is-1.user-profile__username--loading {
  display: block;
  padding-top: 0.4rem;
  margin-bottom: 2rem;
  font-size: 3rem;
}

.user-profile__text--loading {
  background-color: #f6f6f6;
  color: #f6f6f6;
}

