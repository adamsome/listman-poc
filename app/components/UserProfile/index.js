import React from 'react'

import Media from '../Media'
import Avatar from './Avatar'
import './UserProfile.scss'

const UserProfile = ({ user }) => (
  <div className="user-profile">
    <Media
      left={(
        <Avatar
          username={user.username}
          imgSrc={user.avatar}
        />
      )}
      right={(
        <a className="card-header-icon">
          <span className="icon is-large">
            {/* TODO: Base off instagram's lighter icon */}
            <i className="fa fa-ellipsis-h"></i>
          </span>
        </a>
      )}
    >
      <div>
        <h1 className="title is-1">
          {user.username}
        </h1>
        <p className="subtitle is-5">
          {user.description}
        </p>
      </div>
    </Media>
  </div>
)

export default UserProfile


