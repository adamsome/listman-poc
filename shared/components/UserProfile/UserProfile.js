import React from 'react'

import Media from '../Media'
import Avatar from './Avatar'
import './UserProfile.scss'

// If loading, use a blank user object as the loading indicator
const loadingUser = {
  username:
    <span className="user-profile__text--loading">
      {"_".repeat(15)}
    </span>,
  description:
    <span className="user-profile__text--loading">
      {"_".repeat(110)}
    </span>
}

const renderAvatar = (user, isLoading) => (
  <Avatar
    username={user.username}
    url={user.avatar_url}
    isLoading={isLoading}
  />
)

const renderMenu = () => (
  <a className="card-header-icon">
    <span className="icon is-large">
      {/* TODO: Base off instagram's lighter icon */}
      <i className="fa fa-ellipsis-h"></i>
    </span>
  </a>
)

const renderProfile = (user, isLoading) => (
  <div>
    <h1 className={
        "title is-1" + (isLoading ? " user-profile__username--loading" : "")
      }>
      {user.username}
    </h1>
    <p className="subtitle is-5">
      {user.description}
    </p>
  </div>
)

const UserProfile = ({ user, isLoading }) => {
  const userOrBlank = (isLoading || !user) ? loadingUser : user
  return (
    <div>
      <Media
        left={renderAvatar(userOrBlank, isLoading)}
        right={renderMenu()}
      >
        {renderProfile(userOrBlank, isLoading)}
      </Media>
    </div>
  )
}

export default UserProfile


