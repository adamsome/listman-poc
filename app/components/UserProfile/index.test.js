import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import UserProfile from './index'

it('renders correctly', () => {
  const wrapper = shallow(
    <UserProfile
      user={{
        username: "test_username",
        description: "description of user",
        avatar: "http://path/to/avatar",
      }}
    />
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})

