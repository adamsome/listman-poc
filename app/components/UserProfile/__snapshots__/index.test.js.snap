exports[`test renders correctly 1`] = `
<div
  className="user-profile">
  <Media
    left={
      <UserProfileAvatar
        imgSrc="http://path/to/avatar"
        username="test_username" />
    }
    right={
      <a
        className="card-header-icon">
        <span
          className="icon is-large">
          <i
            className="fa fa-ellipsis-h" />
        </span>
      </a>
    }>
    <div>
      <h1
        className="title is-1">
        test_username
      </h1>
      <p
        className="subtitle is-5">
        description of user
      </p>
    </div>
  </Media>
</div>
`;
